src/peripheral/tpiu.rs

```diff
@@ -148,8 +149,8 @@ impl TPIU {
     /// Reads the supported trace output modes and the minimum size of
     /// the TPIU FIFO queue for trace data.
     #[inline]
-    pub fn swo_supports() -> SWOSupports {
-        let _type = unsafe { (*Self::ptr())._type.read() };
+    pub fn swo_supports(&self) -> SWOSupports {
```

Comment:
Per [this comment](https://github.com/rust-embedded/cortex-m/issues/183#issuecomment-579430492), _should_ self be added? Seems not Needed here

Reply:
Indeed not required. I glossed over that comment. I'll revert it.